actions/recurring_reservation.py

```diff
@@ -103,15 +173,20 @@ def pre_calculate_slots(
                     start_datetime=begin,
                     end_datetime=end,
                     is_reservable=True,
-                    buffer_time_after=self.recurring_reservation.reservation_unit.buffer_time_after,
-                    buffer_time_before=self.recurring_reservation.reservation_unit.buffer_time_before,
+                    buffer_time_after=(
+                        reservation_unit.actions.get_actual_after_buffer(begin) if check_buffers else None
+                    ),
+                    buffer_time_before=(
+                        reservation_unit.actions.get_actual_before_buffer(end) if check_buffers else None
```

Comment:
`begin` and `end` seem to be the wrong way round here, but validation of the buffers (when `check_buffers`) should in any case use the buffer values that are going to be saved on the reservation (in `bulk_create_reservation_for_periods`) -- not the settings default from the reservation unit when creating a (non-seasonal) recurring reservation as a staff user,  when the payload supplies the buffer values (unless reservation_unit.reservation_block_whole_day is set in which case that setting will override what is saved).

Reply:
Fixed.